### PR TITLE
Fixed #32691 -- Made Exact lookup on BooleanFields compare directly to a boolean value on MySQL.

### PR DESCRIPTION
**Pull request by romanvm**
This PR fixes the problem when Exact lookup generates suboptimal queries for BooleanField on MySQL.

### CODE REVIEW

**Comment by charettes** [comment]
Two things here.

1. The changes must be made to the non-GIS MySQL backend
2. I wonder if we should have a dedicated `has_native_boolean_(field|type)` feature flag instead and use it to branch off where appropriate

File: django/db/backends/mysql/operations.py [hunk 1]
```diff
@@ -378,3 +380,13 @@ def lookup_cast(self, lookup_type, internal_type=None):
             ):
                 lookup = 'JSON_UNQUOTE(%s)'
         return lookup
+
+    def conditional_expression_supported_in_where_clause(self, expression):
+        if (
+            isinstance(expression, Col) and
+            isinstance(expression.field, BooleanField)
```
[the remark this answers is not in the document]
Comment (todorvelichkov) [reply django/db/backends/mysql/operations.py]: Sorry for the delay, I've run some tests, it works as described in the tests, but as the tests show it does not work for Filters. I think it should work for Filters too if a better functional index support is desired. I'm talking about this:

```
        qs = Author.objects.annotate(
            wrapped=ExpressionWrapper(Q(alias='a1'), output_field=BooleanField()),
        ).filter(wrapped=True)
        self.assertSequenceEqual(qs, [self.au1])
        self.assertNotIn(' = True', str(qs.query))
```

[the remark this answers is not in the document]
Comment (felixxm) [reply django/db/backends/mysql/operations.py]: @todorvelichkov Updated.

**Comment by felixxm** [comment]
@romanvm @adamchainz Can you take a look at updates? :1st_place_medal:  (I'm not a MySQL user)

File: django/db/backends/mysql/operations.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (todorvelichkov) [reply django/db/backends/mysql/operations.py]: I just tested it, I believe its good now, thank you :)

**Comment by romanvm** [comment]
LGTM

**Pull request merged**